task3/Task3.1.1/Survivors.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Task3._1._1
+{
+    public class Survivors : IEnumerable<int>
+    {
+        private List<int> _people;
+        public int Count => _people.Count;
+        public Survivors(int n)
+        {
+            _people = new List<int>(Enumerable.Range(1, n));
+        }
+
+        public IEnumerator<int> GetEnumerator()
+        {
+            return new SurvivorsEnumerator(_people);
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        public void StartGame(int n)
+        {
+            int round = 0;
+            var enumerator = (SurvivorsEnumerator)GetEnumerator();
+            while (Count >= n)
+            {
+                for (int i = 0; i < n; i++)
+                {
+                    enumerator.MoveNext();
+                }
+                _people.Remove(enumerator.Current);
+                enumerator.GoBackOnStep();
```

Comment:
Maybe just do n-1 steps after removing instead of dancing?)

Reply:
after deleting, the elements are shifted, but the index is not. N-1 will result in an unfair deletion, which is why the index decrement is added

Reply:
@Kart164 I meant that you could do less calls to MoveNext instead of moving iterator back